pkg/controller/podautoscaler/metrics/client.go

```diff
@@ -75,18 +75,15 @@ func (c *resourceMetricsClient) GetResourceMetric(ctx context.Context, resource
 	}
 	var res PodMetricsInfo
 	if container != "" {
-		res, err = getContainerMetrics(metrics.Items, resource, container)
-		if err != nil {
-			return nil, time.Time{}, fmt.Errorf("failed to get container metrics: %v", err)
-		}
+		res = getContainerMetrics(ctx, metrics.Items, resource, container)
```

Comment:
Could we document this new behavior in the interface here?

https://github.com/kubernetes/kubernetes/blob/0c5e832aa2bb32573ca695858bc421d111c90119/pkg/controller/podautoscaler/metrics/interfaces.go#L41